sinatra/sinatra_app.rb

```diff
@@ -67,6 +76,51 @@
   dataset[id: id].to_json
 end
 
+post '/api/templates/:id/delete' do
+  protected!
+
+  id = params[:id].to_i
+  dataset = database[:templates]
+  dataset.filter(id: id).delete
+end
+
+post '/api/templates' do
+  protected!
+
+  data = JSON.parse(request.body.read)
+
+  example_activities = "{}"
+  if data['example_activities'].any?
+    example_activities = Sequel.pg_array(data['example_activities'])
+  end
+
+  req_certifications = "{}"
+  if data['req_certifications'].any?
+    req_certifications = Sequel.pg_array(data['req_certifications'])
+  end
+
+  dataset = database[:templates]
+  dataset.insert(
+    job_title: data['job_title'],
+    company_description: data['company_description'],
+    job_description: data['job_description'],
+    req_occupational_skills: Sequel.pg_json(data['req_occupational_skills']),
+    req_foundational_skills: Sequel.pg_json(data['req_foundational_skills']),
+    pref_occupational_skills: Sequel.pg_json(data['pref_occupational_skills']),
+    pref_foundational_skills: Sequel.pg_json(data['pref_foundational_skills']),
+    example_activities: example_activities,
+    req_certifications: req_certifications
+  )
```

Comment:
So, another thing I noticed locally was this request returning with an HTTP status that I think == the new template's `id` (I was seeing 2, 3, 4 as statuses). In ruby, the last line of the method is implicitly returned, and if `dataset.insert()` returns the `id` of the newly inserted row, that would explain it. I know those aren't real response codes, but I don't know if it's an issue. But I wonder if it would help with the heroku error?